src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -87,25 +101,38 @@ public class DocumentDatabase : IResourceStore
 
         public DocumentReplicationLoader DocumentReplicationLoader { get; private set; }
 
-	    public HttpJsonRequestFactory HttpRequestFactory
-	    {
-		    get { return _httpJsonRequestFactory; }
-	    }
+	    public HttpJsonRequestFactory HttpRequestFactory => _httpJsonRequestFactory;
 
 
-	    public void Initialize()
-        {
+	    public void Initialize(HttpJsonRequestFactory httpRequestFactory, DocumentConvention convention)
+	    {
+		    _convention = convention;
+		    _httpJsonRequestFactory = httpRequestFactory;
             DocumentsStorage.Initialize();
             InitializeInternal();
         }
 
-        public void Initialize(StorageEnvironmentOptions options)
+        public void Initialize(StorageEnvironmentOptions options, HttpJsonRequestFactory httpRequestFactory, DocumentConvention convention)
         {
-            DocumentsStorage.Initialize(options);
+			_httpJsonRequestFactory = httpRequestFactory;
+			_convention = convention;
+			DocumentsStorage.Initialize(options);
             InitializeInternal();
         }
 
-        private void InitializeInternal()
+	    public TcpConnectionInfo GetTcpInfo(string url, string apiKey)
```

Comment:
Why is this here? It is only in replication, and it is showing up as a first class item in the DocumentDatabase